packages/yarnpkg-core/sources/httpUtils.ts

```diff
@@ -312,3 +331,37 @@ async function requestImpl(target: string | URL, body: Body, {configuration, hea
     return gotClient(url);
   });
 }
+
+/* @internal */
+export function createAlgoliaClient(configuration: Configuration) {
+  const requester: AlgoliaTypes.Requester = {
+    async send(req: AlgoliaTypes.Request): Promise<AlgoliaTypes.Response> {
+      try {
+        const res = await request(req.url, req.data || null, {
+          configuration,
+          headers: req.headers,
+        });
+
+        return {
+          content: res.body,
+          isTimedOut: false,
+          status: res.statusCode,
+        };
+      } catch (error) {
+        return {
+          content: error.response.body,
+          isTimedOut: false,
+          status: error.response.statusCode,
+        };
+      }
+    },
+  };
+
+  // Note that the appId and appKey are specific to Yarn's plugin-typescript - please
+  // don't use them anywhere else without asking Algolia's permission
+  const ALGOLIA_API_KEY = `e8e1bd300d860104bb8c58453ffa1eb4`;
+  const ALGOLIA_APP_ID = `OFCNCOG2CU`;
+
+  const algoliasearch = require(`algoliasearch`) as typeof import('algoliasearch').default;
+  return algoliasearch(ALGOLIA_APP_ID, ALGOLIA_API_KEY, {requester});
+}
```

Comment:
This doesn't seem like something that belongs in the core.

Reply:
It's reused across a couple of plugins (two). The function is marked internal, so it's not part of the types / public API.